lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -1551,4 +1565,12 @@ private function generateIdentifierName($identifier)
 
         return strtoupper(dechex(crc32($identifier->getName())));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isValidIdentifier($name)
+    {
+        return !!preg_match('/^(?:(?:\p{L}|[_@#])(?:\p{L}|\p{Nd}|[$@#_])*|".*"|\[.*\])$/ui', $name);
```

Comment:
@zeroedin-bill ok, that may or may not be a problem too ;-)

Anyway, can you put comments on what you assume an example (complex) matched identifier would look like? Some things such as the `.*` are a bit fishy imo.